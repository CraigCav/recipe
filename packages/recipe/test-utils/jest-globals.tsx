import React from 'react';
import {render, cleanup} from '@testing-library/react';
import {ThemeProvider} from 'emotion-theming';
import {standard} from '../themes';

import '@testing-library/jest-dom/extend-expect';

const renderWithTheme = renderFn => (component, ...rest) =>
  renderFn(<ThemeProvider theme={standard}>{component}</ThemeProvider>, rest);

export const renderToHtml = renderWithTheme((ui, options) => {
  const {container} = render(ui, options);
  return container.outerHTML;
});

export const fullRender = renderWithTheme(render);

afterEach(cleanup);
