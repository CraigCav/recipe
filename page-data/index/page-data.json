{"componentChunkName":"component---src-pages-index-js","path":"/","webpackCompilationHash":"356c1bd9f0f7a983bfed","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}