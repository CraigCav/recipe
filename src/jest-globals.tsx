import {shallow as enzymeShallow, render as enzymeRender, mount as enzymeMount} from 'enzyme';
import React from 'react';
import {create as reactCreate} from 'react-test-renderer';
import {render as rtlRender, cleanup} from 'react-testing-library';
import {renderToStaticMarkup} from 'react-dom/server';
import {ThemeProvider} from 'emotion-theming';
import {standard} from './themes';

import '@testing-library/jest-dom/extend-expect';

const renderWithTheme = renderFn => (component, ...rest) =>
  renderFn(<ThemeProvider theme={standard}>{component}</ThemeProvider>, rest);

export const shallow = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeShallow(tree, {context});
};

export const mount = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeMount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

export const renderToHtml = renderWithTheme(renderToStaticMarkup);

export const render = renderWithTheme(enzymeRender);

export const fullRender = renderWithTheme(rtlRender);

export const create = renderWithTheme(reactCreate);

afterEach(cleanup);
