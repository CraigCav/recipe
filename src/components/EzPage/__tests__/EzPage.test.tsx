import React from 'react';
import {axe} from 'jest-axe';
import {visualSnapshots} from 'sosia';
import markdown from '../EzPage.md';
import {EzPage, EzPageSection} from '..';
import {renderToHtml} from '../../../jest-globals';
import {EzCard, EzHeading} from '../../index';

const scope = {EzPage, EzPageSection, EzCard, EzHeading};

describe('EzPage', () => {
  visualSnapshots({markdown, scope});

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(
      <EzPage>
        <EzPageSection use="aside">
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzPageSection>
        <EzPageSection use="main">
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzPageSection>
      </EzPage>
    );
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
