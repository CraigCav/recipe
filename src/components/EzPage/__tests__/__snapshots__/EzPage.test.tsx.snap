// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzPage should render EzPageSection content with right sidebar styles 1`] = `
.recipe-4 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-4 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-4 p {
  margin: 0;
}

.recipe-4 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-4 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-4 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-2 {
    width: 290px;
  }
}

@media screen and (min-width:48em) {
  .recipe-2 {
    display: inline-block;
    vertical-align: top;
  }

  .recipe-2 + .recipe-1 {
    margin-left: 3.4285714285714284rem;
  }
}

.recipe-2 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-2 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-2 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-0 {
    width: calc(100% - 290px - 3.4285714285714284rem);
  }
}

@media screen and (min-width:48em) {
  .recipe-0 {
    display: inline-block;
    vertical-align: top;
  }

  .recipe-0 + .recipe-1 {
    margin-left: 3.4285714285714284rem;
  }
}

.recipe-0 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-0 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-0 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

<div
  className="recipe-4 recipe-5"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
  <div
    className="recipe-2 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPage should render EzPageSection with left sidebar styles 1`] = `
.recipe-4 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-4 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-4 p {
  margin: 0;
}

.recipe-4 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-4 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-4 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-0 {
    width: 290px;
  }
}

@media screen and (min-width:48em) {
  .recipe-0 {
    display: inline-block;
    vertical-align: top;
  }

  .recipe-0 + .recipe-1 {
    margin-left: 3.4285714285714284rem;
  }
}

.recipe-0 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-0 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-0 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-2 {
    width: calc(100% - 290px - 3.4285714285714284rem);
  }
}

@media screen and (min-width:48em) {
  .recipe-2 {
    display: inline-block;
    vertical-align: top;
  }

  .recipe-2 + .recipe-1 {
    margin-left: 3.4285714285714284rem;
  }
}

.recipe-2 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-2 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-2 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

<div
  className="recipe-4 recipe-5"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
  <div
    className="recipe-2 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPage should render with default styles 1`] = `
.recipe-0 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-0 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-0 p {
  margin: 0;
}

.recipe-0 > *:not(:last-child) {
  margin-bottom: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-0 > *:not(:last-child) {
    margin-bottom: 1.7142857142857142rem;
  }
}

.recipe-0 > .e1xv9tqn0 {
  margin-bottom: 0.8571428571428571rem;
}

<div
  className="recipe-0 recipe-1"
>
  <p>
    Lorem ipsum dolor
  </p>
  <p>
    Lorem ipsum dolor
  </p>
</div>
`;
